LoopKit/SettingsStore.swift

```diff
@@ -135,7 +180,7 @@ public struct StoredSettings {
     public let basalRateSchedule: BasalRateSchedule?
     public let insulinSensitivitySchedule: InsulinSensitivitySchedule?
     public let carbRatioSchedule: CarbRatioSchedule?
-    public let bloodGlucoseUnit: HKUnit?
```

Comment:
+1

Reply:
As it is not possible to make `HKUnit` implement `Codable` as-is, then I must either: 1) convert it to a `Codable` type (e.g. `String`) upon creation, as I've done, OR, 2) manually implement `Codable` for the entire `StoredSettings` struct (which is a bunch of hand written code that could add errors). I choose the less risky (IMO) option.

@ps2 @rickpasetto If you need me to change it, then I will. Please let me know.

Reply:
Blech, `HKUnit` isn't `Codable` already?  Didn't know that.  It's too bad we use `HKUnit` for this instead of our own type, which we could make `Codable`.  But I imagine that'd be a huge refactor.
@darinkrauss I don't feel that strongly.  I'm fine with what you have.

Reply:
@darinkrauss Just an idea: Could a compromise here be to take in an `HKUnit` in the constructor, and still represent it here as a `String`?   
 You could also have the public `get`ter  return an `HKUnit`, but the `Codable` implementation uses the "internal" String? That way the `init` API remains unchanged, hiding the implementation detail?

Reply:
Yes, that would work. What do you think of that @ps2 ?

Reply:
Munging the data model to satisfy some artifact of a chosen serialization format seems like the tail wagging the dog here. Implementing Codable for this seems like it would be a standard practice, if default encoding wasn't sufficient for some reason. Is it really that hard or risky? Seems like a clean separation of concerns to me.

Reply:
Okay, I will implement full `Codable` for `StoredSettings` so we can have `HKUnit` here.

I do disagree that this is the tail wagging the dog, however. The reason `StoredSettings` exists at all is to be serialized and uploaded.

Reply:
Updated.